hypothesis-auto: wheel/poetry/version-constraints (for streamdeck-ui)

### Steps To Reproduce
Steps to reproduce the behavior:
1. build *streamdeck-ui*

### Build log
```
building '/nix/store/prcbm0zz0cp3zqrxmwc0g18h0llqn8qk-python3.10-hypothesis-auto-1.1.4.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pip-build-hook
Using pipBuildPhase
Using pipShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
unpacking sources
unpacking source archive /nix/store/s4g816a8sid5wim6hkm41y411kgvgv16-hypothesis-auto-1.1.4.tar.gz
source root is hypothesis-auto-1.1.4
setting SOURCE_DATE_EPOCH to timestamp 1580716884 of file hypothesis-auto-1.1.4/PKG-INFO
patching sources
configuring
no configure script, doing nothing
building
Executing pipBuildPhase
Creating a wheel...
WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
Processing /build/hypothesis-auto-1.1.4
  Running command Preparing metadata (pyproject.toml)
  Traceback (most recent call last):
    File "/nix/store/1r39lid0bbn4r49cga7zq95fnv1jrsh5-python3.10-pip-22.2.2/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
      main()
    File "/nix/store/1r39lid0bbn4r49cga7zq95fnv1jrsh5-python3.10-pip-22.2.2/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "/nix/store/1r39lid0bbn4r49cga7zq95fnv1jrsh5-python3.10-pip-22.2.2/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 164, in prepare_metadata_for_build_wheel
      return hook(metadata_directory, config_settings)
    File "/nix/store/vx9yygl10jp63pvh4nlxgvkzvxkb7yl6-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages/poetry/core/masonry/api.py", line 40, in prepare_metadata_for_build_wheel
      poetry = Factory().create_poetry(Path(".").resolve(), with_groups=False)
    File "/nix/store/vx9yygl10jp63pvh4nlxgvkzvxkb7yl6-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages/poetry/core/factory.py", line 64, in create_poetry
      package = self.configure_package(
    File "/nix/store/vx9yygl10jp63pvh4nlxgvkzvxkb7yl6-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages/poetry/core/factory.py", line 158, in configure_package
      cls._add_package_group_dependencies(
    File "/nix/store/vx9yygl10jp63pvh4nlxgvkzvxkb7yl6-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages/poetry/core/factory.py", line 104, in _add_package_group_dependencies
      cls.create_dependency(
    File "/nix/store/vx9yygl10jp63pvh4nlxgvkzvxkb7yl6-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages/poetry/core/factory.py", line 371, in create_dependency
      dependency = Dependency(name, constraint, groups=groups)
    File "/nix/store/vx9yygl10jp63pvh4nlxgvkzvxkb7yl6-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages/poetry/core/packages/dependency.py", line 66, in __init__
      self.constraint = constraint  # type: ignore[assignment]
    File "/nix/store/vx9yygl10jp63pvh4nlxgvkzvxkb7yl6-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages/poetry/core/packages/dependency.py", line 110, in constraint
      self._constraint = parse_constraint(constraint)
    File "/nix/store/vx9yygl10jp63pvh4nlxgvkzvxkb7yl6-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages/poetry/core/constraints/version/parser.py", line 33, in parse_constraint
      constraint_objects.append(parse_single_constraint(and_constraints[0]))
    File "/nix/store/vx9yygl10jp63pvh4nlxgvkzvxkb7yl6-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages/poetry/core/constraints/version/parser.py", line 147, in parse_single_constraint
      raise ParseConstraintError(f"Could not parse version constraint: {constraint}")
  poetry.core.constraints.version.exceptions.ParseConstraintError: Could not parse version constraint: >=0.32.2<2.0.0
  error: subprocess-exited-with-error

  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /nix/store/fkcl1wzq3106qqgl84bhgk1lp56q6bzg-python3-3.10.7/bin/python3.10 /nix/store/1r39lid0bbn4r49cga7zq95fnv1jrsh5-python3.10-pip-22.2.2/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /build/tmp9z1csp0q
  cwd: /build/hypothesis-auto-1.1.4
  Preparing metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
error: builder for '/nix/store/prcbm0zz0cp3zqrxmwc0g18h0llqn8qk-python3.10-hypothesis-auto-1.1.4.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/vpsmpz4qmz05fxi0b7grc4fvq1byxri0-streamdeck-ui-2.0.6.drv' failed to build
error: 1 dependencies of derivation '/nix/store/lpm0fas97bm6nmkmq42ffaffdq6flynm-hwdb.bin.drv' failed to build
error: 1 dependencies of derivation '/nix/store/p79j730ckxs8d95d7w4d4dnlavybp0ak-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/j87qig2av4p7lzlsby1siwc3frxghssy-unit-systemd-udevd.service.drv' failed to build
error: 1 dependencies of derivation '/nix/store/96hm3kdndilk3xrgl46wr2rxd7lbr3nm-nixos-system-XXX-22.11pre-git.drv' failed to build
Build failed: Child process exited with error code: 100
       | Failed: Child process exited with error code: 100
[ERROR] Failed to build XXX - Last 20 lines of logs:
[ERROR]   stderr)   ╰─> See above for output.
[ERROR]   stderr)
[ERROR]   stderr)   note: This error originates from a subprocess, and is likely not a problem with pip.
[ERROR]   stderr)   full command: /nix/store/fkcl1wzq3106qqgl84bhgk1lp56q6bzg-python3-3.10.7/bin/python3.10 /nix/store/1r39lid0bbn4r49cga7zq95fnv1jrsh5-python3.10-pip-22.2.2/lib/python3.10/site-packages/pip/_vendor/pep517/in_process/_in_process.py prepare_metadata_for_build_wheel /build/tmp9z1csp0q
[ERROR]   stderr)   cwd: /build/hypothesis-auto-1.1.4
[ERROR]   stderr)   Preparing metadata (pyproject.toml) ... error
[ERROR]   stderr) error: metadata-generation-failed
[ERROR]   stderr)
[ERROR]   stderr) × Encountered error while generating package metadata.
[ERROR]   stderr) ╰─> See above for output.
[ERROR]   stderr)
[ERROR]   stderr) note: This is an issue with the package mentioned above, not pip.
[ERROR]   stderr) hint: See above for details.
[ERROR]   stderr) error: builder for '/nix/store/prcbm0zz0cp3zqrxmwc0g18h0llqn8qk-python3.10-hypothesis-auto-1.1.4.drv' failed with exit code 1
[ERROR]   stderr) error: 1 dependencies of derivation '/nix/store/vpsmpz4qmz05fxi0b7grc4fvq1byxri0-streamdeck-ui-2.0.6.drv' failed to build
[ERROR]   stderr) error: 1 dependencies of derivation '/nix/store/lpm0fas97bm6nmkmq42ffaffdq6flynm-hwdb.bin.drv' failed to build
[ERROR]   stderr) error: 1 dependencies of derivation '/nix/store/p79j730ckxs8d95d7w4d4dnlavybp0ak-system-path.drv' failed to build
[ERROR]   stderr) error: 1 dependencies of derivation '/nix/store/j87qig2av4p7lzlsby1siwc3frxghssy-unit-systemd-udevd.service.drv' failed to build
[ERROR]   stderr) error: 1 dependencies of derivation '/nix/store/96hm3kdndilk3xrgl46wr2rxd7lbr3nm-nixos-system-XXX-22.11pre-git.drv' failed to build
[ERROR]  failure) Child process exited with error code: 100
[ERROR] Failed to complete requested operation - Last 1 lines of logs:
[ERROR]  failure) Child process exited with error code: 100
[ERROR] -----
[ERROR] Operation failed with error: Child process exited with error code: 100

```

### Additional context
I'm unsure why this happens and how to fix it so I would need help on this

### Notify maintainers
@Majiir 
@jonringer 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.19.16, NixOS, 22.11 (Raccoon), 22.11pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(user): `""`
 - channels(root): `""`
 - nixpkgs: `/run/nixpkgs`
```
(unstable - f994293d1eb8812f032e8919e10a594567cf6ef7)

--- a/pkgs/development/python-modules/hypothesis-auto/default.nix
+++ b/pkgs/development/python-modules/hypothesis-auto/default.nix
@@ -1,8 +1,11 @@
-{ lib, buildPythonPackage, fetchPypi
+{ lib
+, buildPythonPackage
+, fetchPypi
 , hypothesis
 , poetry
 , pydantic
 , pytest
+, pythonOlder
 }:
 
 buildPythonPackage rec {
@@ -10,10 +13,19 @@ buildPythonPackage rec {
   version = "1.1.4";
   format = "pyproject";
 
+  disabled = pythonOlder "3.6";
+
   src = fetchPypi {
     inherit pname version;
-    sha256 = "1c9jksza0gg2gva3liy0s8riv6imjavhnqw05m8l5660knq2yb2y";
+    hash = "sha256-XiwvsJ3AmEJRLYBjC7eSNZodM9LARzrUfuI9oL6eMrE=";
   };
+
+  postPatch = ''
+    # https://github.com/timothycrosley/hypothesis-auto/pull/20
+    substituteInPlace pyproject.toml \
+      --replace 'pydantic = ">=0.32.2<2.0.0"' 'pydantic = ">=0.32.2, <2.0.0"' \
+      --replace 'hypothesis = ">=4.36<6.0.0"' 'hypothesis = "*"'
+  '';
 
   nativeBuildInputs = [
     poetry
@@ -25,7 +37,9 @@ buildPythonPackage rec {
     pytest
   ];
 
-  pythonImportsCheck = [ "hypothesis_auto" ];
+  pythonImportsCheck = [
+    "hypothesis_auto"
+  ];
 
   meta = with lib; {
     description = "Enables fully automatic tests for type annotated functions";

